src/server/common.h

```diff
@@ -255,6 +255,24 @@ class Context : public Cancellation {
   Context(ErrHandler err_handler) : Cancellation{}, err_handler_{std::move(err_handler)} {
   }
 
+  operator GenericError() {
+    return err_;
+  }
+
+  operator std::error_code() {
+    return err_.GetError();
+  }
+
+  // Cancelling the internal context is only possible through the context directly,
+  // because it needs to emit an cancellation error.
+  operator const Cancellation*() {
```

Comment:
I did not understand this

Reply:
Yeah, I probably need to document better the whole context/cancellation code

The context is an extension to cancellation that allows reporting errors and running an error handler that can do cleanup

Cancellation is just a wrapper around an atomic flag.

However, I need to run a cleanup handler for cancellation on the context. If I expose the parent class, then I can't intercept a cancellation request and run the handler, so I either:
- Introduce a cancellation handler on the cancellation (how to combine it with a context)
- Merge Context & Cancellation at all, so they are just one class and will be used everywhere the same (possible solution)
- Don't allow getting a mutable Cancellation from the context, so that it full controls cancellation. This way, it will issue a cancelled error on the cancellation request that will run the error handler and allow doing cleanup